src/interactive/results.ts

```diff
@@ -0,0 +1,170 @@
+import * as vscode from 'vscode'
+
+export interface ResultContent {
+    isIcon: boolean,
+    content: string,
+    hoverContent: string | vscode.MarkdownString,
+    isError: boolean
+}
+
+export class Result {
+    editor: vscode.TextEditor
+    text: string
+    range: vscode.Range
+    content: ResultContent
+    decoration: vscode.TextEditorDecorationType
+    destroyed: boolean
+
+    constructor (editor: vscode.TextEditor, range: vscode.Range, content: ResultContent) {
+        this.editor = editor
+        this.range = range
+        this.text = editor.document.getText(this.range)
+        this.destroyed = false
+
+        this.setContent(content)
+    }
+
+    setContent (content: ResultContent) {
+        if (this.destroyed) {
+            return
+        }
+
+        this.content = content
+
+        if (this.decoration) {
+            this.remove()
+        }
+
+        const color = new vscode.ThemeColor(content.isError ? 'editorError.foreground' : 'editor.foreground')
+
+        let decoration = {
+            after: {
+                contentIconPath: undefined,
+                contentText: undefined,
+                backgroundColor: new vscode.ThemeColor('editorWidget.background'),
+                margin: '0 0 0 10px',
+                color: color
```

Comment:
It seems like the popular extension code lens's inline view has the precedence over:
![image](https://user-images.githubusercontent.com/40514306/82567221-09dd3080-9bb8-11ea-818b-74f5f2eda364.png)

I tried `textDecoration: 'z-index: some_high_number'` here, but without any succeed.

Reply:
I don't think we can really do anything about this. https://github.com/Microsoft/vscode/issues/18666 means there can only ever be one decoration with `after` text for a given range. 

We could switch to an empty range at the end of the line and use `before` instead, but that makes the internals here much more complicated because we now need to keep track of two ranges.

Reply:
makes sense.

>  use `before` instead,

do you mean use `before` here ?:
https://github.com/julia-vscode/julia-vscode/blob/b57c94d4ff5fca5b9dd0f389575cd37471c881cb/src/interactive/results.ts#L41
(it looked really un-intuitively though...)

Reply:
Yes.